File: cs50/week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    float l = count_letters(text);
    float w = count_words(text);
    float s = count_sentences(text);

    // Letters per 100 words
    float L = (l / w) * 100;

    // Sentences per 100 words
    float S = (s / w) * 100;

    // Coleman-Liau Index
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
}

int count_letters(string text)
{
    int n = strlen(text);
    int l = 0;
    for (int i = 0; i < n; i++)
    {
        if (isalpha(text[i]))
        {
            l += 1;
        }
    }
    return l;
}

int count_words(string text)
{
    int n = strlen(text);
    int w = 0;
    if (n == 0)
    {
        return 0;
    }
    else
    {
        for (int i = 0; i < n; i++)
        {
            if (isblank(text[i]))
            {
                w += 1;
            }
        }
    }
    return w + 1;
}

int count_sentences(string text)
{
    int n = strlen(text);
    int s = 0;
    for (int i = 0; i < n; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            s += 1;
        }
    }
    return s;
}

File: cs50/week6/sentimental-readability/readability.py
#!/usr/bin/env python3
def main():

    # TODO: Prompt the user for some text
    while True:
        text = input("Enter text: ")
        if count_letters(text) == 0:
            print("No letters found")
        else:
            break

    # TODO: Count the number of letters, words, and sentences in the text
    letter_count = count_letters(text)
    word_count = count_words(text.strip())
    sentence_count = count_sentences(text)

    # TODO: Compute the Coleman-Liau index
    # Coleman-Liau variables L & S
    L = (letter_count / word_count) * 100
    S = (sentence_count / word_count) * 100
    Coleman_Liau_index = 0.0588 * L - 0.296 * S - 15.8

    # TODO: Print the grade level
    if Coleman_Liau_index > 16:
        print("Grade 16+")
    elif Coleman_Liau_index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(Coleman_Liau_index)}")


def count_letters(text):
    letter_count = 0
    for letter in text:
        if letter.isalpha() == True:
            letter_count += 1
    return letter_count


def count_words(text):
    word_count = 1
    for letter in text:
        if letter == " ":
            word_count += 1
    return word_count


def count_sentences(text):
    sentence_count = 0
    for letter in text:
        if letter == "." or letter == "?" or letter == "!":
            sentence_count += 1
    return sentence_count


main()
